contracts/Acct.sol

```diff
@@ -35,72 +35,112 @@ contract Acct is Ownable, IAcct {
+        // ascertain available balance
+        address self = address(this); // workaround for a possible solidity bug
+        uint256 assetBalance = self.balance;
+        _withdrawETH(assetBalance);
     }
 
     /**
      * @dev Withdraw ETH
      * @param amount Amount of asset to withdraw
      */
     function withdrawETH(uint256 amount) external override onlyOwner isUnlocked {
-        uint256 assetBalance;
+        _withdrawETH(amount);
+    }
 
+    /**
+     * @dev Withdraw ETH (internal)
+     * @param amount Amount of asset to withdraw
+     */
+    function _withdrawETH(uint256 amount) internal {
+        require(amount > 0, "Amount must be greater than zero");
         // ascertain available balance
         address self = address(this); // workaround for a possible solidity bug
-        assetBalance = self.balance;
+        uint256 assetBalance = self.balance;
 
-        // handle withdraw-all
-        if (amount == uint256(-1)) {
-            amount = assetBalance;
-        }
-        require(amount <= assetBalance, "Asset amount < Asset balance");
+        // the following line is not strictly necessary, the transfer would fail anyways
+        require(amount <= assetBalance, "Transfer amount exceeds balance");
 
         // transfer to owner
-        msg.sender.transfer(amount);
+        address payable msgSender = _msgSender();
+        msgSender.transfer(amount);
 
         // log activity
-        emit LogWithdraw(msg.sender, address(0), assetBalance);
+        emit LogWithdraw(msgSender, address(0), amount);
+    }
+
+    /**
+     * @dev Withdraw the total balance of an ERC20 asset.
+     * @param _assetAddress Asset to be withdrawn.
+     */
+    function withdrawAllERC20(address _assetAddress) external override onlyOwner isUnlocked {
+        // ascertain available balance
+        uint256 assetBalance = ERC20(_assetAddress).balanceOf(address(this));
+        _withdrawERC20(_assetAddress, assetBalance);
     }
 
     /**
      * @dev Withdraw ERC20 asset.
      * @param _assetAddress Asset to be withdrawn.
      * @param amount Amount of asset to withdraw
      */
-    function withdrawErc20(address _assetAddress, uint256 amount) external override onlyOwner isUnlocked {
-        uint256 assetBalance;
+    function withdrawERC20(address _assetAddress, uint256 amount) external override onlyOwner isUnlocked {
+        _withdrawERC20(_assetAddress, amount);
+    }
 
+    /**
+     * @dev Withdraw ERC20 asset (internal).
+     * @param _assetAddress Asset to be withdrawn.
+     * @param amount Amount of asset to withdraw
+     */
+    function _withdrawERC20(address _assetAddress, uint256 amount) internal {
+        require(amount > 0, "Amount must be greater than zero");
         // ascertain available balance
-        assetBalance = ERC20(_assetAddress).balanceOf(address(this));
+        uint256 assetBalance = ERC20(_assetAddress).balanceOf(address(this));
```

Comment:
The elimination of the -1 path implies that the withdraw-all case now calls `balanceOf()` repeatedly in the same code execution path.

Reply:
True - however, the downstream check (L101 or L141) isn't actually necessary as in both cases the transfer will fail if the amount exceeds the balance. I will remove the downstream checks.